### PR TITLE
Quarkus Distribution release notes

### LINKED ISSUE #1478: Quarkus distribution release notes for Keycloak 18
### Description

create them.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by DGuhr**
Closes #1478

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]
Awesome, tks.

File: release_notes/topics/18_0_0.adoc [hunk 1]
```diff
@@ -50,9 +50,34 @@ There are more WebAuthn improvements and fixes in addition to that.
...
+
+The {project_name} Quarkus distribution now supports expanding values in keycloak.conf from environment variables, e.g. `log-level=${ROOT_LOG_LVL:info}` would look for an environment variable names `ROOT_LOG_LVL` first and default to info if it could not find it.
+
+For more information, check the corresponding https://www.keycloak.org/server/configuration[guide].
+
+== Quarkus: New Option db-url-port
+
+You can now change the port of your jdbc connection string explicitly by setting the new `db-url-port` configuration option. As for the other convenience options, this option will be overridden by the value of a full `db-url`, if set.
+
+== Quarkus: Split metrics-enabled option into health-enabled and metrics-enabled
```
Comment (vmuzikar) [inline release_notes/topics/18_0_0.adoc]: Should we perhaps mention this in [migration guide](https://github.com/keycloak/keycloak-documentation/blob/main/upgrading/topics/keycloak/changes-18_0_0.adoc) too? It is a breaking change.

Comment (DGuhr) [reply release_notes/topics/18_0_0.adoc]: good point. added it to the migration guide too. :)

File: upgrading/topics/keycloak/changes-18_0_0.adoc [hunk 2]
```diff
@@ -120,3 +120,6 @@ Components known to have breaking changes:
   
   - `noPadding` to `hasNoPadding`
 
+= Quarkus: Split metrics-enabled option into health-enabled and metrics-enabled
+
+The `metrics-enabled` option now only enables the metrics for Keycloak. To enable the readiness and liveness health endpoints, there's a new boolean option `health-enabled`. This allows more fine-grained usage of these options, e.g. enabling metrics but not enabling readiness/liveness probes for on-premise use cases. In order to keep the same behaviour as before when `metrics-enabled=true` was set, now also set `health-enabled=true` when building keycloak.
```
Comment (vmuzikar) [inline upgrading/topics/keycloak/changes-18_0_0.adoc]:
```suggestion
The `metrics-enabled` option now only enables the metrics for {project_name}. To enable the readiness and liveness health endpoints, there's a new boolean option `health-enabled`. This allows more fine-grained usage of these options, e.g. enabling metrics but not enabling readiness/liveness probes for on-premise use cases. In order to keep the same behaviour as before when `metrics-enabled=true` was set, you need to additionally set `health-enabled=true` when building {project_name}.
```

Comment (DGuhr) [reply upgrading/topics/keycloak/changes-18_0_0.adoc]: @vmuzikar could you then change your suggestion to use project_name also in the first sentence? ;)

Comment (DGuhr) [reply upgrading/topics/keycloak/changes-18_0_0.adoc]: btw: this file relies under the "keycloak" folder and there's a different one for rhsso, so I thought I didn't need to change it and looked up others that are also using static "Keycloak"..but whatever works for you.

Comment (vmuzikar) [reply upgrading/topics/keycloak/changes-18_0_0.adoc]: > @vmuzikar could you then change your suggestion to use project_name also in the first sentence? ;)

Done. Good catch. ;)

> btw: this file relies under the "keycloak" folder and there's a different one for rhsso, so I thought I didn't need to change it and looked up others that are also using static "Keycloak"..but whatever works for you.

Good point, but I think just to keep it consistent we should use the var. It is [already used](https://github.com/keycloak/keycloak-documentation/blob/3ddd5c2cfbe99b538806d4fc3d34929e9850930a/upgrading/topics/keycloak/changes-18_0_0.adoc#openid-connect-logout) in that file even.

**Review by vmuzikar — DISMISSED** [review]

File: release_notes/topics/18_0_0.adoc [hunk 3]
```diff
@@ -50,9 +50,33 @@ There are more WebAuthn improvements and fixes in addition to that.
 For more details, see link:{adminguide_link}#_user_session_limits[{adminguide_name}].
 Thanks to https://github.com/mfdewit[Mauro de Wit] for the contribution.
 
+== Quarkus: Import realms at startup
```
Comment (stianst) [inline release_notes/topics/18_0_0.adoc]:
```suggestion
== Quarkus distribution

=== Import realms at startup
```

File: release_notes/topics/18_0_0.adoc [hunk 4]
```diff
@@ -50,9 +50,33 @@ There are more WebAuthn improvements and fixes in addition to that.
...
+
+The {project_name} Quarkus distribution now supports importing your realms directly at start-up. For more information, check the corresponding https://www.keycloak.org/server/importExport[guide].
+
+== Quarkus: JSON and File Logging improvements
```
Comment (stianst) [inline release_notes/topics/18_0_0.adoc]:
```suggestion
=== JSON and File Logging improvements
```

File: release_notes/topics/18_0_0.adoc [hunk 5]
```diff
@@ -50,9 +50,33 @@ There are more WebAuthn improvements and fixes in addition to that.
...
+
+The {project_name} Quarkus distribution now initially supports logging to a File and logging structured data using JSON.
+
+For more information on the improvements, check the corresponding https://www.keycloak.org/server/logging[guide].
+
+== Quarkus: Environment variable expansion for values in keycloak.conf
```
Comment (stianst) [inline release_notes/topics/18_0_0.adoc]:
```suggestion
=== Environment variable expansion for values in keycloak.conf
```

File: release_notes/topics/18_0_0.adoc [hunk 6]
```diff
@@ -50,9 +50,33 @@ There are more WebAuthn improvements and fixes in addition to that.
...
+
+The {project_name} Quarkus distribution now supports expanding values in keycloak.conf from environment variables, e.g. `log-level=${ROOT_LOG_LVL:info}` would look for an environment variable names `ROOT_LOG_LVL` first and default to info if it could not find it.
+
+For more information, check the corresponding https://www.keycloak.org/server/configuration[guide].
+
+== Quarkus: New Option db-url-port
```
Comment (stianst) [inline release_notes/topics/18_0_0.adoc]:
```suggestion
=== New Option db-url-port
```

File: release_notes/topics/18_0_0.adoc [hunk 7]
```diff
@@ -50,9 +50,33 @@ There are more WebAuthn improvements and fixes in addition to that.
...
+
+You can now change the port of your jdbc connection string explicitly by setting the new `db-url-port` configuration option. As for the other convenience options, this option will be overridden by the value of a full `db-url`, if set.
+
+== Quarkus: Split metrics-enabled option into health-enabled and metrics-enabled
```
Comment (stianst) [inline release_notes/topics/18_0_0.adoc]:
```suggestion
=== Split metrics-enabled option into health-enabled and metrics-enabled
```

File: release_notes/topics/18_0_0.adoc [hunk 8]
```diff
@@ -50,9 +50,33 @@ There are more WebAuthn improvements and fixes in addition to that.
...
+The `metrics-enabled` option now only enables the metrics for {project_name}. To enable the readiness and liveness probe, there's the new build option `health-enabled`. This allows more fine-grained usage of these options.
+
 == Other improvements
 
 * Account console alignments with latest PatternFly release.
 * Support for encrypted User Info endpoint response. Thanks to https://github.com/giacomoa[Giacomo Altiero]
 * Support for the algorithm RSA-OAEP with A256GCM used for encryption keys. Thanks to https://github.com/fbrissi[Filipe Bojikian Rissi]
 * Support for login with GitHub Enterprise server. Thanks to https://github.com/nngo[Neon Ngo]
+* Multiple bugfixes and smaller improvements for the Quarkus distribution (e.g. hostname settings, start-up performance)
```
Comment (stianst) [inline release_notes/topics/18_0_0.adoc]: Small improvements/bugfixes are listed in the full release notes so don't need a separate mention here

```suggestion
```

File: upgrading/topics/keycloak/changes-18_0_0.adoc [hunk 9]
```diff
@@ -120,3 +120,6 @@ Components known to have breaking changes:
   
   - `noPadding` to `hasNoPadding`
 
+= Quarkus: Split metrics-enabled option into health-enabled and metrics-enabled
```
Comment (stianst) [inline upgrading/topics/keycloak/changes-18_0_0.adoc]:
```suggestion
= Quarkus distribution: Split metrics-enabled option into health-enabled and metrics-enabled
```

**Review by stianst — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**